viewer/packages/3d-overlays/src/Overlay3DCollection.ts

```diff
@@ -2,123 +2,131 @@
  * Copyright 2023 Cognite AS
  */
 
-import { CanvasTexture, Color, Texture, Vector3, Object3D } from 'three';
+import { CanvasTexture, Color, Texture, Object3D } from 'three';
 import { Overlay3DIcon } from './Overlay3DIcon';
+import { Overlay3D } from './Overlay3D';
 import { OverlayPointsObject } from './OverlayPointsObject';
 import { IconOctree } from './IconOctree';
+import { DefaultMetadataType, OverlayCollection, OverlayInfo } from './OverlayCollection';
 
-export type Overlay3DOptions = {
+export type Overlay3DCollectionOptions = {
   overlayTexture?: Texture;
   maxPointSize?: number;
+  defaultOverlayColor?: Color;
 };
 
-export type PointData = {
-  position: Vector3;
-  id: number;
-  color?: Color;
-};
+export class Overlay3DCollection<MetadataType = DefaultMetadataType>
+  extends Object3D
+  implements OverlayCollection<MetadataType>
+{
+  private readonly MinPixelSize = 16;
+  private readonly MaxPixelSize = 64;
+  private readonly DefaultMaxPoints = 100000;
```

Comment:
Are these "true" constants (e.g. they will never be anything else than the values set here)? If so, perhaps we can just put them outside the class and let them be all-caps.